api/krusty/accumulation_test.go

```diff
@@ -167,12 +169,35 @@ spec:
 
 func TestAccumulateResourcesErrors(t *testing.T) {
 	type testcase struct {
-		name     string
-		resource string
-		// regex error message that is output when kustomize tries to
-		// accumulate resource as file and dir, respectively
+		name       string
+		resource   string
+		isAbsolute bool
+		files      map[string]string
+		// errFile, errDir are regex for the expected error message output
+		// when kustomize tries to accumulate resource as file and dir,
+		// respectively. The test substitutes occurrences of "%s" in the
+		// error strings with the absolute path where kustomize looks for it.
 		errFile, errDir string
 	}
+	populateAbsolutePaths := func(tc testcase, dir string) testcase {
+		filePaths := make(map[string]string, len(tc.files)+1)
```

Comment:
I think this `len` and range of `tc.files` are the same numerics.
Are you need `+1` for `len(tc.files)` for any reason?

Reply:
I moved this line of code: https://github.com/kubernetes-sigs/kustomize/pull/5225/files#diff-15a9eb3d3d62fd27ecea240ff711d887220b80ac54f3f709a7ed8bc25632a837R195 Does this make it easier to understand?

In addition to all the files we need to create specified by the test case, we also need to create the build root or the direct kustomization that we run `kustomize build` on.

Reply:
Sorry, This is my misunderstanding.
Thanks for explaining that!